subprojects/core/src/integTest/groovy/org/gradle/api/invocation/GradleRootProjectIntegrationTest.groovy

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.invocation
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+class GradleRootProjectIntegrationTest extends AbstractIntegrationSpec {
+
+    def 'settings plugin from included build can apply plugin by id to the root project'() {
+        file("build-logic/src/main/kotlin/foo.gradle.kts") << """
+            println("Foo applied")
+        """
+
+        file("build-logic/src/main/kotlin/bar.settings.gradle.kts") << """
+            gradle.rootProject {
+                plugins.apply("foo")
+            }
+        """
+
+        buildFile("build-logic/build.gradle.kts", """
```

Comment:
💅 Since it's kts, the Groovy build script highlighting complains in the IDE. I suggest to use the `<<` to avoid that for now. Also, we usually drop the parenthesis around the arguments

Reply:
@alllex Sorry, read it couple of times and didn't get the idea.
For the `buildFile` API what is the difference kts/groovy? Sort of for kts we using `<<` (and effectivelt there is no need for `buildFile` usage then, just `file` is enough) and for groovy we do use `buildFile "foo", "bar"`?

Reply:
The `buildFile` function annotates the argument with the code snippet and provides Groovy highlighting for it. If Kotlin (script) source is passed as an argument, IDEA starts to complain in yellow or red that something is wrong, because it tries to parse Groovy. It does not break the test in any way, because it's only the in-IDE experience, but I would avoid misleading the IDE.

That's why I suggest to avoid using a function that tries to highlight Groovy for the test that exercises Kotlin scripts.

We don't have a sibling function for `buildFile` that would have highlighted Kotlin. It might be non-trivial to set up, but that should be tried separately.